app/src/main/java/com/x8bit/bitwarden/ui/platform/manager/snackbar/SnackbarRelayManagerImpl.kt

```diff
@@ -0,0 +1,36 @@
+package com.x8bit.bitwarden.ui.platform.manager.snackbar
+
+import com.x8bit.bitwarden.data.platform.repository.util.bufferedMutableSharedFlow
+import com.x8bit.bitwarden.ui.platform.components.snackbar.BitwardenSnackbarData
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.onCompletion
+
+/**
+ * The default implementation of the [SnackbarRelayManager] interface.
+ */
+class SnackbarRelayManagerImpl : SnackbarRelayManager {
+    private val mutableSnackbarRelayMap =
+        mutableMapOf<SnackbarRelay, MutableSharedFlow<BitwardenSnackbarData?>>()
+
+    override fun sendSnackbarData(data: BitwardenSnackbarData, relay: SnackbarRelay?) {
```

Comment:
Does this need to be a nullable `SnackbarRelay`

Reply:
Ahh, now I see why you did that. Not sure how I feel about it

Reply:
Instead of the UUIDs, how would you feel about an enum that identified each specific snackbar that requires a relay.

Then you subscribe to all the values you need and you publish to a specific enum value as well.

Reply:
I feel like that is the move, may have tried to go a little too agnostic, but no reason why producer and consumer can't know exactly what they are doing when sending/collecting. Great idea!